lua/registers.lua

```diff
@@ -272,27 +272,15 @@ local function apply_register(register)
 
 	-- Handle insert mode differently
 	if invocation_mode == "i" then
-		-- Handle the special case for the expression register
-		if register == "=" then
-			-- Get the proper keycode for <C-R>
-			local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
+		-- Get the proper keycode for <C-R>
+		local key = vim.api.nvim_replace_termcodes("<c-r>", true, true, true)
 
-			-- Opening the expression register in normal mode magically works
-			-- I have no idea why..
+		if register == "=" then
 			vim.api.nvim_feedkeys(key .. "=", "n", true)
+		else
+			local submit = vim.api.nvim_replace_termcodes("<CR>", true, true, true)
```

Comment:
Can you add comments describing what it does similar to the rest?

Reply:
sure